### PR TITLE
SOT-177: Add Montgomery term

### PR DESCRIPTION
**Pull request by VanhGer**
# Check List

- [x] Did you check your grammar / reader proof your document using ChatGPT?
- [x] Don't forget to squash commits into meaningful chunks before merging
- [x] After merging this, please update your task status!
- [x] Did you verify that your math formulas are rendered correctly?

### CODE REVIEW

File: terms/montgomery_reduction.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+Step 5. x := MonPro(¯x, 1) 
+Step 6. return x
+```
+
+### Example
+
+We show how to compute $9 * 11 \mod 13$ ($= 8$). We have:
+
+- $r = 2^4 = 16$
+- Using the Extended Euclid algorithm, we determine that $16 ** 9 − 13 * 11 = 1$, thus, $r^{-1} = 9, n' = 11$
```
Comment (zk-steve) [inline terms/montgomery_reduction.md:67]:
```suggestion
- Using the Extended Euclid algorithm, we determine that $16 * 9 − 13 * 11 = 1$, thus, $r^{-1} = 9, n' = 11$
```

File: terms/montgomery_reduction.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+Step 7. x := MonPro(¯x, 1) 
+Step 8. return x
+```
+
+### Example
+
+We show how to compute $x = 7^{10} \mod 13$ ($=4$). We have:
+
+- $r = 2^4 = 16$
+- Using the Extended Euclid algorithm, we determine that $16 ** 9 − 13 * 11 = 1$, thus, $r^{-1} = 9, n' = 11$
```
Comment (zk-steve) [inline terms/montgomery_reduction.md:112]:
```suggestion
- Using the Extended Euclid algorithm, we determine that $16 * 9 − 13 * 11 = 1$, thus, $r^{-1} = 9, n' = 11$
```

File: terms/montgomery_reduction.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+To compute this product, we need to determine $r^{-1}$, which satisfies the property:  
+
+$r \cdot r^{-1} = 1 \mod n$
+$\iff r \cdot r^{-1} - n \cdot n' = 1$
+
+Therefore, we can find $r^{-1}$ and $n'$ using the
+[Extended Euclid algorithm](https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm). We can express $\bar{R}$ as follows:
+
+$\bar R = \bar a \cdot \bar b \cdot r^{-1} \mod n$
```
Comment (zk-steve) [inline terms/montgomery_reduction.md:33]: use array to align equations

File: terms/montgomery_reduction.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+purposes, we suppose that $n$ is an odd number.
+
+## Preliminaries
+
+First, we choose $r > n$ such that: $(r, n) = 1$. Because $n$ is odd, we often choose $r = 2^k$.
+
+This technique replaces division by $n$ with division by a power of $2$, , which is easily performed on a computer since numbers are
+represented in binary form.
+
+To achieve this, we need to convert all the numbers we use into Montgomery space. The $n$-residue of $a$ in Montgomery space is given
```
[the remark this answers is not in the document]
Comment (VanhGer) [reply terms/montgomery_reduction.md:16]: achieve the replacement

File: terms/montgomery_reduction.md (hunk 3 above)
Comment (VanhGer) [reply terms/montgomery_reduction.md:33]: I dont know how to use that? Please give me a link to the example

**Review by zk-steve — APPROVED** [review]

**Pull request merged**